src/UI/Art/SpriteArtist/CommanderOverlayArtist.java

```diff
@@ -0,0 +1,248 @@
+      g.drawImage( powerBarImage, 0, (overlayImage.getHeight() * drawScale) - (POWERBAR_BUFFER*drawScale), powerBarImage.getWidth() * drawScale, powerBarImage.getHeight() * drawScale, null );
+    }
+    else
+    { // Draw the overlay on the right side.
+      int screenWidth = SpriteOptions.getScreenDimensions().width;
+      int xPos = screenWidth - overlayImage.getWidth() * drawScale;
+      int coNameXPos = screenWidth - spriteA.getWidth() * drawScale * coString.length() - xTextOffset;
+      int fundsXPos = screenWidth - spriteA.getWidth() * drawScale * overlayFundsString.length() - xTextOffset;
+      g.drawImage(overlayImage, xPos, 0, overlayImage.getWidth() * drawScale, overlayImage.getHeight() * drawScale, null);
+      SpriteLibrary.drawTextSmallCaps(g, coString, coNameXPos, yTextOffset, drawScale); // CO name
+      SpriteLibrary.drawTextSmallCaps(g, overlayFundsString, fundsXPos, textHeight + drawScale + yTextOffset, drawScale); // Funds
+      int pbXPos = screenWidth - powerBarImage.getWidth() * drawScale;
+      if( trueIfBarFalseIfText )
+      {
+        // We are drawing the power bar, and want to flip it horizontally.
+        g.drawImage( powerBarImage, pbXPos + (powerBarImage.getWidth() * drawScale), (overlayImage.getHeight() * drawScale) - (POWERBAR_BUFFER*drawScale), -powerBarImage.getWidth() * drawScale, powerBarImage.getHeight() * drawScale, null );
+      }
+      else
+      {
+        // We are drawing an ability name, and we don't want to flib it.
+        g.drawImage( powerBarImage, pbXPos, (overlayImage.getHeight() * drawScale) - (POWERBAR_BUFFER*drawScale), powerBarImage.getWidth() * drawScale, powerBarImage.getHeight() * drawScale, null );
+      }
+    }
+  }
+
+  private static void updateAnimIndex()
+  {
+    // Calculate the sprite index to use.
+    long thisTime = System.currentTimeMillis();
+    long animTimeDiff = thisTime - animIndexUpdateTime;
+
+    // If it's time to update the sprite index... update the sprite index.
+    if( animTimeDiff > animIndexUpdateInterval )
+    {
+      animIndex++;
+      animIndexUpdateTime = thisTime;
+    } 
+  }
+
+  /**
+   * Constructs a fixed-width (padded as needed) 11-character string to be drawn in the commander overlay.
+   * @param funds The number to convert to an HUD overlay funds string.
+   * @return A string of the form "FUNDS XXXXX" where X is either a space or a digit.
+   */
+  private static String buildFundsString(Commander cmdr)
+  {
+    StringBuilder sb = new StringBuilder("FUNDS ");
+    if( cmdr.money < 10000 ) // Fewer than 5 digits
+    {
+      sb.append(" ");
+    }
+    if( cmdr.money < 1000 ) // Fewer than 4 digits
+    {
+      sb.append(" ");
+    }
+    if( cmdr.money < 100 ) // Fewer than 3 digits
+    {
+      sb.append(" ");
+    }
+    if( cmdr.money < 10 ) // Fewer than 2 digits. You poor.
+    {
+      sb.append(" ");
+    }
+    sb.append(Integer.toString(cmdr.money));
+
+    return sb.toString();
```

Comment:
I feel like there's a string format function to produce a fixed width result, and I don't like excess if statements.

Reply:
Fixed.